docs/guide/configuration.md

```diff
@@ -53,7 +55,9 @@ And now you'll be able to write queries like this:
 const query = myDb.rawQuery(sql`SELECT * FROM users`);
 ```
 
-## 2. A basic example of migrations folder
+## Example 2: Migrations with Shadow Database
+
+To avoid having to keep your database up to date with your schema manually, SafeQL also offers a way to synchronize a "shadow database" with your `.sql` migration files with the [`migrationsDir` option](https://safeql.dev/api/index.html#connections-migrationsdir).
```

Comment:
Omit "with Shadow Database"

Most of the devs that use migrations aren't aware of the idea of a "shadow database", and they don't have to. Shadow database in SafeQL is an ad-hoc solution for fetching the database metadata since we can't run queries against the migrations folder.

I have an open issue (https://github.com/ts-safeql/safeql/issues/8) that should make it easier for newcomers. Only developers who stumble across this technique issue will need to get a not-so-deep dive into it.

Reply:
Ok, I have a version that goes in this direction here: 0433587dee3892a11943126494ac34d36ecbb47f

Good enough?

Reply:
One last thing - there are many good reasons why we use migrations aside from avoiding manually keeping the database up to date. Maybe we should replace that sentence with something like "of you have a migration folder ..."

Reply:
So I wanted to provide a clear reason why migrations are important \*\*for SafeQL\*\*. I don't mean to describe why migrations are important as a concept generally.

I wanted to answer the question (the same question that I had when I saw SafeQL for the first time):

> Migrations, what? Why would SafeQL need configuration for migrations?

Reply:
We discussed offline and landed at a description like this:

> Eliya: we could write something "if you have a migration folder, you might want to choose this option instead of databaseUrl since ...
> Eliya: And something that the queries will be in sync with the current state of the code, which is more accurate

I'll take a look at this

Reply:
I have a version of this in 6827fd55c59e205ced1b60cb3895ff7066912ed2, what do you think?